consensus/misc/eip2935.go

```diff
@@ -0,0 +1,42 @@
+package misc
+
+import (
+	// "fmt"
+	"math/big"
+
+	"github.com/holiman/uint256"
+
+	"github.com/ledgerwatch/erigon-lib/chain"
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/consensus"
+	"github.com/ledgerwatch/erigon/core/state"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/params"
+)
+
+func StoreBlockHashesEip2935(header *types.Header, state *state.IntraBlockState, config *chain.Config, headerReader consensus.ChainHeaderReader) {
+	if header.Number.Cmp(big.NewInt(0)) == 0 { // Activation of fork at Genesis
+		return
+	}
+	parent := headerReader.GetHeaderByHash(header.ParentHash)
```

Comment:
Better use the faster and more reliable `GetHeader` instead of `GetHeaderByHash` since we know the right number.